python/cudf/cudf/core/indexed_frame.py

```diff
@@ -3929,7 +3929,7 @@ def groupby(
                 "squeeze parameter is not yet implemented"
             )
 
-        if observed is not False:
+        if observed is False:
```

Comment:
Test this as a "falsy" object rather than checking object identity with `is`.
```suggestion
        if not observed:
```